packages/react-components/src/components/ActionMenu/ActionMenu.tsx

```diff
@@ -39,23 +43,70 @@ export interface ActionMenuProps {
    * Menu will stay open after option click
    */
   keepOpenOnClick?: boolean;
+  /**
+   * Set the menu placement to keep it in view
+   */
+  flipOptions?: Parameters<typeof flip>[0];
+  /**
+   * Set to control the menu visibility
+   */
+  visible?: boolean;
+  /**
+   * Optional handler called on menu close
+   */
+  onClose?: () => void;
+  /**
+   * Optional handler called on menu open
+   */
+  onOpen?: () => void;
 }
 
 const baseClass = 'action-menu';
 
 export const ActionMenu: React.FC<ActionMenuProps> = ({
   className,
-  triggerClassName,
   options,
   triggerRenderer,
   placement = 'bottom-end',
   openedOnInit = false,
   keepOpenOnClick,
+  flipOptions,
+  visible,
+  onClose,
+  onOpen,
   ...props
 }) => {
+  const isControlled = visible !== undefined;
   const [isVisible, setIsVisible] = React.useState(openedOnInit);
   const indexRef = React.useRef<number>(-1);
   const ref = React.useRef<HTMLUListElement | null>(null);
+  const currentlyVisible = isControlled ? visible : isVisible;
+
+  const handleMenuStateChange = () => {
+    if (currentlyVisible) {
+      onClose?.();
+      !isControlled && setIsVisible(false);
+    } else {
+      onOpen?.();
+      !isControlled && setIsVisible(true);
+    }
+  };
+
+  const { x, y, strategy, refs, context } = useFloating({
+    middleware: [offset(4), flip(flipOptions)],
+    placement: placement,
+    open: currentlyVisible,
+    onOpenChange: handleMenuStateChange,
+    whileElementsMounted: autoUpdate,
+  });
+  const click = useClick(context);
+  const dismiss = useDismiss(context, {
```

Comment:
This hook have options that can be useful for other devs ([props](https://floating-ui.com/docs/useDismiss#props)). I think we can make an optional prop with this options available publicly.